### PR TITLE
wallet2: factor the watchonly/multisig/etc fields on creation

### PR DESCRIPTION
**Pull request by moneromooo-monero**
There's half a dozen calls, and it's easy to miss some when
adding a new field.

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -4289,12 +4282,7 @@ void wallet2::restore(const std::string& wallet_, const epee::wipeable_string& p
 
   m_account.create_from_device(hwdev);
   m_key_device_type = m_account.get_device().get_type();
-  m_account_public_address = m_account.get_keys().m_account_address;
-  m_watch_only = false;
-  m_multisig = false;
-  m_multisig_threshold = 0;
-  m_multisig_signers.clear();
-  m_original_keys_available = false;
+  init_type(m_key_device_type); // weird
```
Comment (stoffu) [inline src/wallet/wallet2.cpp]: Is the comment `weird` in the sense that it ends up doing `m_key_device_type = m_key_device_type;`? If so, I'd suggest removing the `device_type` arg from `init_type` and set the default of `m_key_device_type` to `device_type::SOFTWARE`. Or maybe even better to set the default to `m_account.get_device().get_type()`?

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: It means I found it a bit weird that this particular one was not setting m_key_device_type. It might be a bug.

Comment (stoffu) [reply src/wallet/wallet2.cpp]: I see `m_key_device_type = m_account.get_device().get_type();` just above the deleted lines. Does it seem weird to you?

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: Hmm. No. In fact, I managed to miss it somehow. I'll change it to pass it to the function.

**Pull request merged**